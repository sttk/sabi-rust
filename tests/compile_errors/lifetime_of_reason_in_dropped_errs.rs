use sabi::Err;

#[derive(Debug)]
enum Reasons {
  MyErr { msg: String, flag: bool },
}

fn return_err() -> Result<(), Err> {
  let err = Err::new(Reasons::MyErr { msg: "hello".to_string(), flag: true });
  Err(err)
}

fn consume_err(_err: Err) {}

fn main() {
  let err = return_err().unwrap_err();
  let reason = err.reason::<Reasons>().unwrap();
  println!("{:?}", reason);
  consume_err(err);
  println!("{:?}", reason);
}
