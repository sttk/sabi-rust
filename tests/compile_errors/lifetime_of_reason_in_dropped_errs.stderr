error[E0505]: cannot move out of `err` because it is borrowed
  --> tests/compile_errors/lifetime_of_reason_in_dropped_errs.rs:19:15
   |
16 |   let err = return_err().unwrap_err();
   |       --- binding `err` declared here
17 |   let reason = err.reason::<Reasons>().unwrap();
   |                --- borrow of `err` occurs here
18 |   println!("{:?}", reason);
19 |   consume_err(err);
   |               ^^^ move out of `err` occurs here
20 |   println!("{:?}", reason);
   |                    ------ borrow later used here
